### PR TITLE
Checked legend items refactor

### PR DESCRIPTION
**Pull request by chowington**
Fixes https://github.com/VEuPathDB/web-eda/issues/1370

### CODE REVIEW

**Review by moontrip — COMMENTED** [review]
Works fine overall: tested Vizs and FSM 👍  Just one note that there is a legend bug at FSM, which was addressed by https://github.com/VEuPathDB/web-eda/pull/1362. Thus you may need to check it out carefully when merging.

**Review by bobular — COMMENTED** [review]
I was just about to test this, but then had a phone call, and now have to head to the airport. It's better if I test carefully, and I can also think about the map-specific functionality.

File: src/lib/core/hooks/checkedLegendItemsStatus.ts [hunk 1]
```diff
@@ -2,41 +2,24 @@ import { useCallback, useEffect, useMemo } from 'react';
...
 ) {
   /**
    * Reset checkedLegendItems to all-checked (actually none checked)
    * if ANY of the checked items are NOT in the vocabulary
    * OR if ALL of the checked items ARE in the vocabulary
    *
-   * TO DO: generalise this for use in other visualizations
+   * TO DO: generalise this for use in all visualizations
    */
   useEffect(() => {
```
[the remark this answers is not in the document]
Comment (moontrip) [reply src/lib/core/hooks/checkedLegendItemsStatus.ts:24]: @bobular Aha I see. Good to know 👍 Safe trip!

**Review by bobular — APPROVED** [review]
I've tested now and all is well.

I have noticed something we can tidy up now or leave to another ticket.

Note that the `fullScreenMap` does not reset `checkedLegendItems` to `undefined` when changing overlay variables, while all other visualizations (including MapVisualization) do this.

What this means in the FSM is that if you choose an overlay variable, check some boxes, and then deactivate the overlay variable, and then reactivate the SAME overlay variable, your checkbox selections will be preserved.

In all other visualizations, changing or cancelling the overlay variable will always reset `checkedLegendItems`.

In FSM, the only thing that will reset `checkedLegendItems` is a change to a NEW overlayvariable that has different values for the variable. The code that does the resetting is [here](https://github.com/VEuPathDB/web-eda/blob/51df2a222aeb54c60cf88daa22ac9a7a558225d6/src/lib/core/hooks/checkedLegendItemsStatus.ts#L28-L37).  So in SCORE S. mansoni Cluster Randomized Trial, you can choose overlay variable "Schistosoma mansoni, by microscopy" and have only "Positive" selected in the checkboxes.  If you change to another positive/negative variable, like "Hookworm, by microscopy" your checkbox selections will be preserved.

I think it's better to reset the checked legend items on every overlay variable change, In the example above "Positive" means roughly the same thing between different overlay variables, but for other variables, especially "yes/no" variables, I'm not sure we'd want to preserve the checkboxes.

I'll commit the small change to fullScreenMap.tsx to make this happen.

**Comment by chowington** [comment]
> Works fine overall: tested Vizs and FSM 👍 Just one note that there is a legend bug at FSM, which was addressed by #1362. Thus you may need to check it out carefully when merging.

Looks like this is working as expected!

Thanks @moontrip and @bobular for helping out!

**Pull request merged**